firestore/google/cloud/firestore_v1beta1/_helpers.py

```diff
@@ -816,23 +843,24 @@ def remove_server_timestamp(document_data):
     actual_data = {}
     for field_name, value in six.iteritems(document_data):
         if isinstance(value, dict):
-            sub_field_paths, sub_data = remove_server_timestamp(value)
-            field_paths.extend(
-                get_field_path([field_name, sub_path])
-                for sub_path in sub_field_paths
-            )
+            sub_field_paths, sub_data = remove_server_timestamp(value, False)
+            for sub_path in sub_field_paths:
+                field_path = FieldPath.from_string(field_name)
+                field_path.parts = field_path.parts + sub_path.parts
+                field_paths.extend([field_path])
             if sub_data:
                 # Only add a key to ``actual_data`` if there is data.
                 actual_data[field_name] = sub_data
         elif value is constants.SERVER_TIMESTAMP:
-            field_paths.append(field_name)
+            if top_level:
```

Comment:
Is `top_level` really `split_on_dots`?

I am a little bit concerned by your usage of strings in this method. `field_name` (which is really a field path) should be a FieldPath class so that this method can be called from an update method that takes FieldPaths.

Reply:
I think only updates are `split_on_dots` on top level, but sets aren't.  I'll take a look and maybe rename that better.

This was written before the FieldPath class, so I think it does take strings as the default.

Reply:
You are correct that 'split_on_dots' only happens at the top level. You can always pass false when you invoke this function recursively.

Reply:
so would you like me to change the name `top_level` to `split_on_dots` or is `top_level` better?

Reply:
I think `split_on_dots` is much clearer in its intend.